### PR TITLE
Someone likes my posted/shared article notification

### PR DESCRIPTION
**Pull request by mohamedHedi1990**

### CODE REVIEW

File: services/src/main/java/org/exoplatform/news/notification/plugin/LikeNewsNotificationPlugin.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    super(initParams);
+    this.newsService = newsService;
+    this.spaceService = spaceService;
+    this.sessionProviderService = sessionProviderService;
+    this.identityManager = identityManager;
+  }
+
+  @Override
+  public String getId() {
+    // TODO Auto-generated method stub
```
Comment (thomasdelhomenie) [inline services/src/main/java/org/exoplatform/news/notification/plugin/LikeNewsNotificationPlugin.java]: Line to remove.

File: services/src/main/java/org/exoplatform/news/notification/plugin/LikeNewsNotificationPlugin.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    return ID;
+  }
+
+  @Override
+  public boolean isValid(NotificationContext ctx) {
+    // TODO Auto-generated method stub
```
Comment (thomasdelhomenie) [inline services/src/main/java/org/exoplatform/news/notification/plugin/LikeNewsNotificationPlugin.java]: Line to remove.

File: services/src/main/java/org/exoplatform/news/notification/plugin/LikeNewsNotificationPlugin.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                               .with(NotificationConstants.ILLUSTRATION_URL, illustrationUrl)
+                               .with(NotificationConstants.ACTIVITY_LINK, activityLink)
+                               .with(NotificationConstants.CONTEXT, context)
+                               .key(getKey())
+                               .end();
+      }
+
+    } catch (Exception e) {
+      // TODO Auto-generated catch block
+      e.printStackTrace();
```
Comment (thomasdelhomenie) [inline services/src/main/java/org/exoplatform/news/notification/plugin/LikeNewsNotificationPlugin.java]: Do not use printStackTrace().

File: webapp/src/main/webapp/WEB-INF/conf/news/notification/configuration.xml [hunk 4]
```diff
@@ -157,6 +157,78 @@
...
+              <string>LikeNewsNotificationPlugin</string>
+            </field>
+            <field name="order">
+              <string>4</string>
+            </field>
+            <field name="defaultConfig">
+              <collection type="java.util.ArrayList">
+                <value>
+                  <string>daily</string>
+                </value>
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/WEB-INF/conf/news/notification/configuration.xml]: Do we have this notification in the daily digest ?

File: services/src/main/java/org/exoplatform/news/notification/plugin/LikeNewsNotificationPlugin.java (hunk 1 above)
Comment (mohamedHedi1990) [reply services/src/main/java/org/exoplatform/news/notification/plugin/LikeNewsNotificationPlugin.java]: Done.

File: services/src/main/java/org/exoplatform/news/notification/plugin/LikeNewsNotificationPlugin.java (hunk 2 above)
Comment (mohamedHedi1990) [reply services/src/main/java/org/exoplatform/news/notification/plugin/LikeNewsNotificationPlugin.java]: Done.

File: services/src/main/java/org/exoplatform/news/notification/plugin/LikeNewsNotificationPlugin.java (hunk 3 above)
Comment (mohamedHedi1990) [reply services/src/main/java/org/exoplatform/news/notification/plugin/LikeNewsNotificationPlugin.java]: Done.

File: webapp/src/main/webapp/WEB-INF/conf/news/notification/configuration.xml (hunk 4 above)
Comment (mohamedHedi1990) [reply webapp/src/main/webapp/WEB-INF/conf/news/notification/configuration.xml]: It is not specified. I did as the other plugin configurations.

Comment (thomasdelhomenie) [reply webapp/src/main/webapp/WEB-INF/conf/news/notification/configuration.xml]: ok. Let's keep it like that. I will ask Sameh if we need to change this globally or not.

**Pull request merged**